src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -2946,6 +2946,26 @@ public bool OpenCustomNodeWorkspace(Guid guid)
 
             return false;
         }
+        /// <summary>
+        ///     Opens an existing custom node workspace.
+        /// </summary>
+        /// <param name="guid">Identifier of the workspace to open</param>
+        /// <returns>True if workspace was found and open</returns>
+        internal bool OpenCustomNodeWorkspaceSilent(Guid guid)
+        {
+            CustomNodeWorkspaceModel customNodeWorkspace;
+            if (CustomNodeManager.TryGetFunctionWorkspace(guid, IsTestMode, out customNodeWorkspace))
+            {
+                if (!Workspaces.OfType<CustomNodeWorkspaceModel>().Contains(customNodeWorkspace))
+                {
+                    AddWorkspace(customNodeWorkspace);
```

Comment:
After the custom node workspace loaded silently, we should also remove it somehow after Python engine switched?

Reply:
I thought about that as well, but the workspaces should/are marked dirty IMO, so user will know which custom nodes were updated. One scenario where this would be bad is maybe if a user has too many custom nodes?
Also saving the custom nodes for them and then closing the workspace seemed a bit hacky, would you like that instead?

Reply:
I would prefer the current option then, maybe when customer has too many dyfs in the canvas, then this could potentially be a problem

Reply:
Yes, we could consult customers for feedback and then later update this functionality.